api/src/services/application-csv-export.service.ts

```diff
@@ -815,15 +698,380 @@ export class ApplicationCsvExporterService
+   * @param queryParams
+   * @param req
+   * @returns generates the lottery export file via helper function and returns the streamable file
+   */
+  async lotteryExport<QueryParams extends ApplicationCsvQueryParams>(
+    req: ExpressRequest,
+    res: Response,
+    queryParams: QueryParams,
+  ): Promise<StreamableFile> {
+    const user = mapTo(User, req['user']);
+    await this.authorizeCSVExport(user, queryParams.listingId);
+
+    const filename = join(
+      process.cwd(),
+      `src/temp/lottery-listing-${queryParams.listingId}-applications-${
+        user.id
+      }-${new Date().getTime()}.csv`,
+    );
+
+    await this.createLotterySheet(filename, queryParams);
+    const file = createReadStream(filename);
+    return new StreamableFile(file);
+  }
+
+  /**
+   *
+   * @param filename
+   * @param queryParams
+   * @returns generates the lottery sheet
+   */
+  async createLotterySheet<QueryParams extends ApplicationCsvQueryParams>(
+    filename: string,
+    queryParams: QueryParams,
+  ): Promise<void> {
+    const applications = await this.prisma.applications.findMany({
+      select: {
+        id: true,
+        preferences: true,
+        householdMember: {
+          select: {
+            id: true,
+          },
+        },
+        applicationLotteryPositions: {
+          select: {
+            ordinal: true,
+            multiselectQuestionId: true,
+          },
+          where: {
+            multiselectQuestionId: null,
+          },
+          orderBy: {
+            ordinal: OrderByEnum.DESC,
+          },
+        },
+      },
+      where: {
+        listingId: queryParams.listingId,
+        deletedAt: null,
+        markedAsDuplicate: false,
+      },
+    });
+
+    // get all multiselect questions for a listing to build csv headers
+    const multiSelectQuestions =
+      await this.multiselectQuestionService.findByListingId(
+        queryParams.listingId,
+      );
+
+    // get maxHouseholdMembers associated to the selected applications
+    let maxHouseholdMembers = 0;
+    applications.forEach((app) => {
+      if (app.householdMember?.length > maxHouseholdMembers) {
+        maxHouseholdMembers = app.householdMember.length;
+      }
+    });
+
+    const csvHeaders = await this.getCsvHeaders(
+      maxHouseholdMembers,
+      multiSelectQuestions,
+      queryParams.timeZone,
+      queryParams.includeDemographics,
+      true,
+    );
+
+    return this.csvExportHelper(
+      filename,
+      mapTo(Application, applications),
+      csvHeaders,
+      queryParams,
+      true,
+    );
+  }
+
+  /**
+   *
+   * @param queryParams
+   * @param req
+   * @returns generates the lottery results for a listing
+   */
+  async lotteryGenerate<QueryParams extends ApplicationCsvQueryParams>(
+    req: ExpressRequest,
+    res: Response,
+    queryParams: QueryParams,
+  ): Promise<SuccessDTO> {
+    const user = mapTo(User, req['user']);
+    await this.authorizeCSVExport(user, queryParams.listingId);
```

Comment:
I could be misremembering. But isn't it by default only admins that are allowed to generate the lottery? So do we need to have a separate check for permissions than this `authorizeCSVExport` endpoint?

Reply:
ah you're right, I thought that the permissions would be the same between export and lottery gen but that shouldn't be the case

I switched this to throw a forbidden exception if the user is not an admin